src/Idephix/Extension/Slack/Slack.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Idephix\Extension\Slack;
+
+use Idephix\IdephixInterface;
+use Idephix\Extension\IdephixAwareInterface;
+
+/**
+ * Description of Slack wrapper
+ *
+ * @author dymissy
+ */
+class Slack implements IdephixAwareInterface
+{
+    /**
+     * @var \Idephix\IdephixInterface
+     */
+    private $idx;
+
+    private $settings;
+
+    public function __construct($args = array())
+    {
+        $defaults = array(
+            'url' => '',
+            'channel' => '#general',
+            'icon_url' => 'https://slack.com/img/icons/app-57.png',
+            'username' => 'slackbot'
+        );
+
+        $this->settings = array_merge($defaults, $args);
+    }
+
+    public function sendToSlack($message, $attachments = array(), $channel = '', $icon_url = '', $username = '')
+    {
```

Comment:
>Are we sure we want to expose all these parameters ? The final user can set all these data in the $config array.

`$attachments` is optional. Only the `message` parameter is required. I've added the full array just as example. `$config` array could be a solution but what if I want to customize different kind of messages?

>What the $attachment parameter is supposed to contain?

`$attachments` contains parameters supported from APIs. 

>1. $attachment is perhaps the wrong name
>2. the data passed is extremely complicated
>3. how the final user is supposed to know the right structure to pass?

1. attachment is the name that Slack uses in its own APIs. I just mantained the name.
2. I'm totally agree with you but I've just used the parameters requested from Slack APIs. Suggestions in order to make it easier?
3. I've just attached Slack documentation links in the annotation of the task and I thought to add a section in Idephix documentation when the PR will be approved.